### PR TITLE
[flink] add set type for mysql cdc action

### PR DESCRIPTION
**Pull request by zhangjun0x01**
*(Please specify the module before the PR name: [core] ... or [flink] ...)*

### Purpose

*related to https://github.com/apache/incubator-paimon/issues/966 *

### Tests

* org.apache.paimon.flink.action.cdc.mysql.MySqlSyncTableActionITCase *

### API and Format 

*(Does this change affect API or storage format)*

### Documentation

*(Does this change introduce a new feature)*

### CODE REVIEW

**Comment by zhangjun0x01** [comment]
image

File: paimon-common/src/main/java/org/apache/paimon/utils/TypeUtils.java [hunk 1]
```diff
@@ -115,6 +118,19 @@ public static Object castFromString(String s, DataType type) {
             case TIMESTAMP_WITHOUT_TIME_ZONE:
                 TimestampType timestampType = (TimestampType) type;
                 return toTimestamp(str, timestampType.getPrecision());
+            case ARRAY:
+                ArrayType arrayType = (ArrayType) type;
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/utils/TypeUtils.java:122]: if str startswith "[", remove "[".
if str endswith "]", remove "]".

**Comment by zhangjun0x01** [comment]
image

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**